nixos/roles/nix-remote-builder.nix

```diff
@@ -0,0 +1,36 @@
+{ lib, config, pkgs, ... }:
+let
+  cfg = config.roles.nix-remote-builder;
+in
+{
+  _file = ./nix-remote-builder.nix;
+
+  options.roles.nix-remote-builder = {
+    schedulerPublicKeys = lib.mkOption {
+      description = "SSH public keys of the central build scheduler";
+      type = lib.types.listOf lib.types.str;
+    };
+  };
+
+  config = {
+    # Garbage-collect often
+    nix.gc.automatic = true;
+    nix.gc.dates = "*:45";
+    nix.gc.options = ''--max-freed "$((128 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))"'';
+
+    # Randomize GC to avoid thundering herd effects.
+    systemd.timers.nix-gc.timerConfig.RandomizedDelaySec = lib.mkForce "1800";
+
+    # Allow more open files for non-root users to run NixOS VM tests.
+    security.pam.loginLimits = [
+      { domain = "*"; item = "nofile"; type = "-"; value = "20480"; }
+    ];
+
+    # Give restricted SSH access to the build scheduler
+    users.extraUsers.root.openssh.authorizedKeys.keys = map
```

Comment:
We should probably not use root to do remote builds

Reply:
Probably. I think there was some performance reason to use root, so it bypasses the nix-daemon entirely. I know that remote building requires trusted users at a minimum.